src/engine/TransitivePathImpl.h

```diff
@@ -153,35 +163,30 @@ class TransitivePathImpl : public TransitivePathBase {
     auto [startSide, targetSide] = decideDirection();
     std::shared_ptr<const Result> subRes = subtree_->getResult();
 
-    IdTable idTable{allocator()};
-
-    idTable.setNumColumns(getResultWidth());
-
-    size_t subWidth = subRes->idTable().numColumns();
-
     if (startSide.isBoundVariable()) {
       std::shared_ptr<const Result> sideRes =
-          startSide.treeAndCol_.value().first->getResult();
-      size_t sideWidth = sideRes->idTable().numColumns();
+          startSide.treeAndCol_.value().first->getResult(true);
 
-      CALL_FIXED_SIZE((std::array{resultWidth_, subWidth, sideWidth}),
-                      &TransitivePathImpl<T>::computeTransitivePathBound, this,
-                      &idTable, subRes->idTable(), startSide, targetSide,
-                      sideRes->idTable());
+      auto gen =
+          computeTransitivePathBound(std::move(subRes), startSide, targetSide,
+                                     std::move(sideRes), !requestLaziness);
 
-      return {std::move(idTable), resultSortedOn(),
-              Result::getMergedLocalVocab(*sideRes, *subRes)};
+      return requestLaziness
+                 ? ProtoResult{std::move(gen), resultSortedOn()}
+                 : ProtoResult{cppcoro::getSingleElement(std::move(gen)),
+                               resultSortedOn()};
     }
-    CALL_FIXED_SIZE((std::array{resultWidth_, subWidth}),
-                    &TransitivePathImpl<T>::computeTransitivePath, this,
-                    &idTable, subRes->idTable(), startSide, targetSide);
+    auto gen = computeTransitivePath(std::move(subRes), startSide, targetSide,
+                                     !requestLaziness);
 
     // NOTE: The only place, where the input to a transitive path operation is
     // not an index scan (which has an empty local vocabulary by default) is the
     // `LocalVocabTest`. But it doesn't harm to propagate the local vocab here
     // either.
-    return {std::move(idTable), resultSortedOn(),
-            subRes->getSharedLocalVocab()};
+    return requestLaziness
```

Comment:
The note above this seems out of place now that the handling of the local vocabs is done somewhere else.